lib/cinematic.ts

```diff
@@ -905,3 +969,66 @@ class Cinematic {
         window.removeEventListener('copy', copy);
     }
 }
+
+class CinematicVideo {
+    poster: string;
+    subtitles: string | null;
+    sources: VideoQuality[];
+
+    constructor(options: VideoOptions) {
+        this.poster = options.poster;
+        this.subtitles = options.subtitles;
+        this.sources = options.sources;
+    }
+
+    getSourcesForQuality(quality: string): VideoQuality | null {
+        if (this.sources.length === 1) {
+            return this.sources[0];
+        }
+        for (let source of this.sources) {
+            if (source.quality === quality) {
+                return source;
+            }
+        }
+        return null;
```

Comment:
does this really make sense to always return null if no value is found and the value if there is only one?

we already need to handle the `null` value anyway --> see `handleQualityChange` calling this method, so we might as well return the best quality in the `null` case to be consistent with the implementation for 1 source